common/analysis/file_analyzer.h

```diff
@@ -88,8 +88,7 @@ std::ostream& operator<<(std::ostream&, const RejectedToken&);
 class FileAnalyzer {
  public:
   explicit FileAnalyzer(absl::string_view contents, absl::string_view filename)
-      : text_structure_(std::make_unique<TextStructure>(contents)),
-        filename_(filename) {}
+      : text_structure_(new TextStructure(contents)), filename_(filename) {}
```

Comment:
I kind of prefer `make_unique` and `make_shared` stylistically, as a way to minimize the direct calls of `operator new` and make it clear on-the-spot what kind of ownership is involved.

Reply:
Problem is, then I have to give general `make_unique<>` friendship to the constructor; but I wanted to be specific in this case to know exactly who is making use of being friends.